core/templates/dev/head/domain/classifier/CountVectorizerService.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Vecotrizer function which mirrors the CountVectorizer feature
+ * extractor of sklearn.
+ *
+ * IMPORTANT NOTE: The Vectorizer function uses the vocabulary that was
+ * extracted during training. During training scikit's
+ * CountVectorizer class is used for this purpose. If there are any changes
+ * in scikit's CountVectorize class then corresponding changes must be
+ * propagated here.
+ */
+
+oppia.factory('CountVectorizerService', [function() {
+  return {
+    vectorize: function(tokens, vocabulary) {
+      var vectorLength = Math.max.apply(Math, Object.values(vocabulary)) + 1;
```

Comment:
Oh I see. I will change it to `Object.key` because we only need to know length, so it does not matter whether we use keys or values. Is that fine?